crates/vhost-user-backend/src/backend.rs

```diff
@@ -110,6 +113,35 @@ pub trait VhostUserBackend: Send + Sync {
         vrings: &[Self::Vring],
         thread_id: usize,
     ) -> Result<()>;
+
+    /// Initiate transfer of internal state for the purpose of migration to/from the back-end.
+    ///
+    /// Depending on `direction`, the state should either be saved (i.e. serialized and written to
+    /// `file`) or loaded (i.e. read from `file` and deserialized).  Note that this function must
+    /// not block during transfer, i.e. I/O to/from `file` must be done outside of this function.
```

Comment:
Maybe we should add documentation on the return type; without reading the vhost-user doc, it isn’t obvious why a `File` can be returned. Something like “The back-end can choose to use a different channel than `file`. If so, it must return a `File` that the front-end can use.”?

Reply:
Yes, I plan to add more docs (for that reason I leave the [Draft])

Reply:
(Looks good to me in the current version. (github won’t let me resolve the conversation, because only the PR author and anyone with write access to the repo can do so, hence this comment.))